src/theme/themes/utils/generate-theme.scss

```diff
@@ -19,61 +39,76 @@
     "gradient-secondary-start": color.adjust($secondary, $saturation: 4%),
     "gradient-secondary-end": color.adjust($secondary, $saturation: 5%, $lightness: -19%),
   );
+  @if ($page-background != null) {
+    $result: map.set($result, "color-page-background", $page-background);
+  }
   @return $result;
 }
-@mixin generateTheme($p, $s) {
-  $colorPalette: getColorPalette($p, $s);
+@mixin generateTheme($p, $s, $bg: null) {
+  $colorPalette: getColorPalette($p, $s, $bg);
   @include colorVariables($colorPalette);
 }
 
 @mixin colorVariables($sourceColorPalette) {
   $colorPalette: $sourceColorPalette;
 
-  --ion-color-primary: #{map-get($colorPalette, "color-primary")};
-  --ion-color-primary-light: #{map-get($colorPalette, "color-primary-light")};
-  --ion-color-primary-lighter: #{map-get($colorPalette, "color-primary-lighter")};
-  --ion-color-primary-contrast: #{map-get($colorPalette, "color-primary-contrast")};
+  --ion-color-primary: #{map.get($colorPalette, "color-primary")};
+  --ion-color-primary-50: #{map.get($colorPalette, "color-primary-50")};
```

Comment:
nit(non-blocking): I'm pretty sure this can also boil down to a couple-liner, e.g.
```
 @each $name, $color in $colorPalette {
    --ion-#{$name}: #{$color};
  }
```
Might be worth testing out and possibly tidying up